types/agent-base/index.d.ts

```diff
@@ -0,0 +1,44 @@
+// Type definitions for agent-base 4.2
+// Project: https://github.com/TooTallNate/node-agent-base#readme
+// Definitions by: Christopher Quadflieg <https://github.com/Shinigami92>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+import { EventEmitter } from 'events';
+
+declare namespace Agent {
+	type AgentCallback = (
+		req?: any,
+		opts?: {
+			secureEndpoint: boolean;
+		}
+	) => void;
+
+	interface AgentOptions {
```

Comment:
According to this options only have timeout as property : https://github.com/TooTallNate/node-agent-base#agentfunction-callback-object-options--httpagent

Reply:
In this line [TooTallNate/node-agent-base/.../index.js#L41](https://github.com/TooTallNate/node-agent-base/blob/master/index.js#L41) you see that `opts` is passed to `this.options`

And in a line like this [TooTallNate/node-agent-base/.../index.js#L75](https://github.com/TooTallNate/node-agent-base/blob/master/index.js#L75) `opts.host` and `opts.path` are definitely used